### PR TITLE
Request improvements

### PR DESCRIPTION
**Pull request by georgesimms**
* Introduces an hx.text that passes through the responseText rather than an xhr
* Creates a 'hx.reshapedRequest' that determines which type of parser to use based on the Content-Type header

### CODE REVIEW

File: modules/request/main/index.coffee [hunk 1]
```diff
@@ -113,29 +113,33 @@ hx.request = -> # url, callback, options || url, data, callback, options
   hx_xhr urlType, urls, data or null, callback, options
 
 
-reshapedRequest = (args, type, formatter) ->
-  urls = args[0]
+parsers =
+  'application/json': (text) -> if text then JSON.parse text
+  'text/html': (text) -> hx.parseHTML text
+  'text/plain': (text) -> text
```
Comment (georgesimms) [inline modules/request/main/index.coffee]: These parsers operate on the string, not the XHR.

File: modules/request/main/index.coffee [hunk 2]
```diff
@@ -113,29 +113,33 @@ hx.request = -> # url, callback, options || url, data, callback, options
...
-  reshapedRequest arguments, 'application/json', (result) ->
-    if result.responseText then JSON.parse(result.responseText)
-
-hx.html = ->
-  reshapedRequest arguments, 'text/html', (result) ->
-    hx.parseHTML result.responseText
+hx.json = (args...) -> reshapedRequest args, 'application/json'
+hx.html = (args...) -> reshapedRequest args, 'text/html'
+hx.text = (args...) -> reshapedRequest args, 'text/plain'
+hx.reshapedRequest = (args...) -> reshapedRequest args
```
Comment (georgesimms) [inline modules/request/main/index.coffee]: use splat feature of coffeescript

File: modules/request/main/index.coffee [hunk 3]
```diff
@@ -113,29 +113,33 @@ hx.request = -> # url, callback, options || url, data, callback, options
...
+  'text/plain': (text) -> text
 
-  if hx.isFunction(args[1])
-    callback = args[1]
-    options = args[2]
-  else
-    data = args[1]
-    callback = args[2]
-    options = args[3]
+reshapedRequest = ([urls, data, callback, options], type) ->
```
Comment (georgesimms) [inline modules/request/main/index.coffee]: use coffeescript destructuring

File: modules/request/test/spec.coffee [hunk 4]
```diff
@@ -624,7 +624,83 @@ describe 'hx-request', ->
...
+        hx.reshapedRequest 'test.html', cb
+
+        expect(jasmine.Ajax.requests.at(0).method).toEqual('GET')
+        jasmine.Ajax.requests.at(0).respondWith
+          status: 200
+          contentType: 'text/html'
+          responseText: '<div class="parent"><span class="child">bob</span></div>'
+
+    it 'should pass the data through', ->
+      data = {some: 'data'}
```
Comment (georgesimms) [inline modules/request/test/spec.coffee]: On a more general note, is it ok that we assume that the post data is always JSON, irrespective of the mime type of the response ? More traditional servers might expect form-encoded data, for instance.

**Comment by c-frater** [comment]
Fixes #70

**Pull request merged**